mitodl/mitxpro
# #252 Home Page: Upcoming Courses

#### Pre-Flight checklist


#### What are the relevant tickets?
#252 

#### What's this PR do?
Adds the ability to override contents of a `CoursewareCarousel` subpage. This type of subpage can only be added under a `ProgramPage` or a `HomePage`. The page (carousel) by default shows all courses from the associated program. However, there are no associated programs with the `HomePage`. This is where the override comes in. The CMS now has fields for `override_contents` and the actual `contents`. Given `override_contents` is checked, the carousel will display information selected under the `contents` stream and not try to access the associated program. This allows the administrator to manually add as many courses or programs to the carousel as they want, and can also reorder them.

#### How should this be manually tested?
Add a `CoursewareCarousel` subpage to the `HomePage`. Check the `override_contents` box and add some items to the `contents` stream. The carousel should appear as expected on the home page.

#### Screenshots (if appropriate)
![Screenshot from 2019-05-23 14-51-22](https://user-images.githubusercontent.com/45350418/58243687-92897c80-7d6a-11e9-8216-2f4bdb512993.png)
![Screenshot from 2019-05-23 14-51-43](https://user-images.githubusercontent.com/45350418/58243693-94534000-7d6a-11e9-81ff-0001134def18.png)
![Screenshot from 2019-05-23 14-53-12](https://user-images.githubusercontent.com/45350418/58243699-974e3080-7d6a-11e9-9ae5-065b89f121f4.png)
